crescent/plugin.py

```diff
@@ -151,7 +150,7 @@ def _add_plugin(self, path: str, plugin: Plugin, refresh: bool = False) -> None:
         plugin._load(self._client)
 
 
-class Plugin:
+class UserPlugin(Generic[BotT]):
```

Comment:
I think we should just leave the name as `Plugin`

Reply:
If I do that, plugins that are not subclassed have to be declared as `plugin = crescent.Plugin[hikari.GatewayBot]()`. Maybe this is ok though.

Reply:
```py
class BasePlugin(Generic[BotT]):
    # everything here


class Plugin(BasePlugin[GatewayTraits | RESTTraits]):
    pass
```

this would allow the old way to work fine, and just tell people to subclass BasePlugin instead?

Reply:
I really don't like that naming scheme. `UserPlugin` and `Plugin` are more descriptive of the purpose IMO.